extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.js

```diff
@@ -179,13 +186,27 @@ export class AmpLightboxViewer extends AMP.BaseElement {
             const clonedNode = element.cloneNode(deepClone);
             clonedNode.removeAttribute('on');
             const descText = this.manager_.getDescription(element);
-            if (descText) {
-              clonedNode.descriptionText = descText;
+            const metadata = {
+              'descriptionText': descText,
+              'tagName': clonedNode.tagName,
+            };
+
+            if (clonedNode.tagName === 'AMP-IMG') {
+              const container = this.element.ownerDocument.createElement('div');
+              container.classList.add('i-amphtml-image-lightbox-container');
+              const imageViewer = new ImageViewer(this, this.win,
+                this.loadPromise.bind(this));
+              imageViewer.init(element, element);
+              container.appendChild(imageViewer.getElement());
+              this.carousel_.appendChild(container);
+              this.clonedLightboxableElements_.push(container);
+              metadata.imageViewer = imageViewer;
+            } else {
+              this.clonedLightboxableElements_.push(clonedNode);
```

Comment:
nit: refactoring not to repeat this might be good. Maybe a variable named `slide` that either becomes the clonedNode or the imageViewer?

Reply:
Done